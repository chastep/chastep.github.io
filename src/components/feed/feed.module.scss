@use "src/assets/styles/mixins";
@use "src/assets/styles/variables";

.feed {
  .item {
    @include mixins.margin-bottom(1.25);

    &:last-child {
      @include mixins.margin-bottom(0.5);
    }

    .title {
      font-size: variables.$typography-base-font-size * 1.6875;
      @include mixins.line-height(1.5);
      @include mixins.margin-top(0);
      @include mixins.margin-bottom(0.5);

      .link {
        color: hsl(var(--color-foreground));

        &:hover,
        &:focus {
          border-bottom: 1px solid hsl(var(--color-foreground));
          color: hsl(var(--color-foreground));
        }
      }
    }

    .description {
      color: hsl(var(--color-muted-foreground));
      font-size: variables.$typography-base-font-size;
      @include mixins.line-height(1);
      @include mixins.margin-bottom(0.75);
    }

    .meta {
      .time {
        color: hsl(var(--color-foreground));
        font-size: variables.$typography-small-font-size;
        font-weight: 600;
        text-transform: uppercase;
      }

      .divider {
        @include mixins.margin(0, 0.5);
      }

      .category {
        .link {
          font-weight: 600;
          text-transform: uppercase;
          color: hsl(var(--color-primary));
          font-size: variables.$typography-small-font-size;
        }
      }
    }

    .more {
      color: hsl(var(--color-foreground));
      font-size: variables.$typography-base-font-size;

      &:hover,
      &:focus {
        border-bottom: 1px solid hsl(var(--color-primary));
        color: hsl(var(--color-primary));
      }
    }
  }
}
