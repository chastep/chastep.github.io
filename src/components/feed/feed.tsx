import React, { type FC } from "react";

import { Link } from "gatsby";

import { type Edge } from "@/types/edge";

import * as styles from "./feed.module.scss";

type FeedProps = {
  edges: Array<Edge>;
};

const Feed: FC<FeedProps> = ({ edges }) => (
  <div className={styles.feed}>
    {edges.map((edge) => (
      <div className={styles.item} key={edge.node.fields.slug}>
        <div className={styles.meta}>
          <time
            className={styles.time}
            dateTime={new Date(edge.node.frontmatter.date).toLocaleDateString(
              "en-US",
              { year: "numeric", month: "long", day: "numeric" },
            )}
          >
            {new Date(edge.node.frontmatter.date).toLocaleDateString("en-US", {
              year: "numeric",
              month: "long",
            })}
          </time>
          <span className={styles.divider} />
          <span className={styles.category}>
            <Link to={edge.node.fields.categorySlug} className={styles.link}>
              {edge.node.frontmatter.category}
            </Link>
          </span>
        </div>
        <h2 className={styles.title}>
          <Link
            className={styles.link}
            to={edge.node.frontmatter?.slug || edge.node.fields.slug}
          >
            {edge.node.frontmatter.title}
          </Link>
        </h2>
        <p className={styles.description}>
          {edge.node.frontmatter.description}
        </p>
        <Link
          className={styles.more}
          to={edge.node.frontmatter?.slug || edge.node.fields.slug}
        >
          Read
        </Link>
      </div>
    ))}
  </div>
);

export { Feed };
