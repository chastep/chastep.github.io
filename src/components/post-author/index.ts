export { PostAuthor } from "./post-author";
