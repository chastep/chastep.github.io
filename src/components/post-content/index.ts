export { PostContent } from "./post-content";
