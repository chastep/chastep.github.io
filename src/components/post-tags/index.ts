export { PostTags } from "./post-tags";