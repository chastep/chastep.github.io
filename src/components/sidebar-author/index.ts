export { SidebarAuthor } from "./sidebar-author";