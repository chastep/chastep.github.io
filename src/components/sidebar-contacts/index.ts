export { SidebarContacts } from "./sidebar-contacts";
