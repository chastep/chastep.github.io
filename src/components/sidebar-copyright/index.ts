export { SidebarCopyright } from "./sidebar-copyright";
