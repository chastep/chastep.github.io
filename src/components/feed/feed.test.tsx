import React from "react";
import { test, describe, expect } from "bun:test";

import * as mocks from "@/mocks";
import { Feed } from "@/components/feed";
import { createSnapshotsRenderer } from "@/utils/render-with-coil-provider";

describe("Feed", () => {
  test("renders correctly", () => {
    expect(createSnapshotsRenderer(<Feed edges={mocks.edges} />).toJSON()).toMatchSnapshot();
  });
});
