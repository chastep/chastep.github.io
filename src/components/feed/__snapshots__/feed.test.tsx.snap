// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "September 2016",
              ],
              "props": {
                "className": undefined,
                "dateTime": "September 1, 2016",
              },
              "type": "time",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    "Typography",
                  ],
                  "props": {
                    "aria-current": undefined,
                    "className": undefined,
                    "href": "/typography",
                    "onClick": [Function],
                    "onMouseEnter": [Function],
                  },
                  "type": "a",
                },
              ],
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Perfecting the Art of Perfection",
              ],
              "props": {
                "aria-current": undefined,
                "className": undefined,
                "href": "/posts/perfecting-the-art-of-perfection",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "h2",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "An Essay on Typography by Eric Gill takes the reader back to the year 1930. The year when a conflict between two worlds came to its term. The machines of the industrial world finally took over the handicrafts.",
          ],
          "props": {
            "className": undefined,
          },
          "type": "p",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Read",
          ],
          "props": {
            "aria-current": undefined,
            "className": undefined,
            "href": "/posts/perfecting-the-art-of-perfection",
            "onClick": [Function],
            "onMouseEnter": [Function],
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "September 2016",
              ],
              "props": {
                "className": undefined,
                "dateTime": "September 1, 2016",
              },
              "type": "time",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    "Design Inspiration",
                  ],
                  "props": {
                    "aria-current": undefined,
                    "className": undefined,
                    "href": "/design-inspiration",
                    "onClick": [Function],
                    "onMouseEnter": [Function],
                  },
                  "type": "a",
                },
              ],
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Johannes Gutenberg: The Birth of Movable Type",
              ],
              "props": {
                "aria-current": undefined,
                "className": undefined,
                "href": "/posts/johannes-gutenberg",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "h2",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "German inventor Johannes Gutenberg developed a method of movable type and used it to create one of the western world’s first major printed books, the “Forty–Two–Line” Bible.",
          ],
          "props": {
            "className": undefined,
          },
          "type": "p",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Read",
          ],
          "props": {
            "aria-current": undefined,
            "className": undefined,
            "href": "/posts/johannes-gutenberg",
            "onClick": [Function],
            "onMouseEnter": [Function],
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
    {
      "$$typeof": Symbol(react.test.json),
      "children": [
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "September 2017",
              ],
              "props": {
                "className": undefined,
                "dateTime": "September 1, 2017",
              },
              "type": "time",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": null,
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                {
                  "$$typeof": Symbol(react.test.json),
                  "children": [
                    "Typography",
                  ],
                  "props": {
                    "aria-current": undefined,
                    "className": undefined,
                    "href": "/typography",
                    "onClick": [Function],
                    "onMouseEnter": [Function],
                  },
                  "type": "a",
                },
              ],
              "props": {
                "className": undefined,
              },
              "type": "span",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "div",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            {
              "$$typeof": Symbol(react.test.json),
              "children": [
                "Humane Typography in the Digital Age",
              ],
              "props": {
                "aria-current": undefined,
                "className": undefined,
                "href": "/posts/humane-typography-in-the-digital-age",
                "onClick": [Function],
                "onMouseEnter": [Function],
              },
              "type": "a",
            },
          ],
          "props": {
            "className": undefined,
          },
          "type": "h2",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": null,
          "props": {
            "className": undefined,
          },
          "type": "p",
        },
        {
          "$$typeof": Symbol(react.test.json),
          "children": [
            "Read",
          ],
          "props": {
            "aria-current": undefined,
            "className": undefined,
            "href": "/posts/humane-typography-in-the-digital-age",
            "onClick": [Function],
            "onMouseEnter": [Function],
          },
          "type": "a",
        },
      ],
      "props": {
        "className": undefined,
      },
      "type": "div",
    },
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
